bertie_bogo_shim/src/main.rs

```diff
@@ -0,0 +1,214 @@
+        .with_max_level(Level::DEBUG)
+        .init();
+
+    let mut args: Vec<_> = env::args().collect();
+    args.remove(0);
+
+    if let Some(arg) = args.get(0) {
+        if arg == "-is-handshaker-supported" {
+            println!("No");
+            process::exit(0);
+        }
+    }
+
+    println!("Options: {:#?}", args);
+
+    let mut options = Options::default();
+
+    while !args.is_empty() {
+        let arg = args.remove(0);
+        match arg.as_ref() {
+            "-server" => {
+                options.role = Role::Server;
+                skip_currently(&arg);
+            }
+            "-key-file" => {
+                options.key_file = args.remove(0);
+            }
+            "-cert-file" => {
+                options.cert_file = args.remove(0);
+            }
+            "-min-version" => {
+                skip_currently(&arg);
+            }
+            "-max-version" => {
+                skip_currently(&arg);
+            }
+            "-host-name" => {
+                options.hostname = args.remove(0);
+            }
+            "-port" => {
+                options.port = args.remove(0).parse::<u16>().unwrap();
+            }
+            "-fallback-scsv" => {
+                options.expect_fallback_scsv = true;
+                skip_currently(&arg);
+            }
+            "-expect-extended-master-secret" => {
+                options.expect_extended_master_secret = true;
+                skip_currently(&arg);
+            }
+            // -------------------------------------------------------------------------------------
+            "-dtls" | "-quic" | "-no-tls13" => {
+                skip(&arg);
+            }
+            "-no-tls1" | "-no-tls11" | "-no-tls12" => {
+                ignore(&arg);
+            }
+            // -------------------------------------------------------------------------------------
+            "-max-send-fragment" => {
+                skip(&arg);
+            }
+            arg => {
+                if UNHANDLED_ARGUMENTS.contains(&arg) {
+                    skip_currently(arg);
+                }
+
+                println!("Unhandled argument | \"{}\",", arg);
+                process::exit(1);
+            }
+        }
+    }
+
+    println!("{:#?}", options);
+
+    match options.role {
+        Role::Client => {
+            let addrs = [
+                std::net::SocketAddr::from((std::net::Ipv6Addr::LOCALHOST, options.port)),
+                std::net::SocketAddr::from((std::net::Ipv4Addr::LOCALHOST, options.port)),
+            ];
+            let stream = TcpStream::connect(&addrs[..]).expect("Can't connect to BoGo.");
+
+            let _ = tls13client(&options.hostname, stream, "hello");
+        }
+        Role::Server => {
+            unimplemented!()
+        }
+    }
+}
+
+fn skip(due_to: &str) {
+    println!("Skipping test due to \"{}\" argument.", due_to);
+    process::exit(BOGO_NACK);
+}
+
+// TODO: Remove eventually.
```

Comment:
Add an issue if you want to keep the `TODO`.

Reply:
Same as https://github.com/cryspen/bertie/pull/41#discussion_r959408368